test/test-fs.c

```diff
@@ -2751,19 +2751,23 @@ TEST_IMPL(fs_write_alotof_bufs_with_offset) {
 TEST_IMPL(fs_read_write_null_arguments) {
   int r;
 
-  r = uv_fs_read(NULL, NULL, 0, NULL, 0, -1, NULL);
+  r = uv_fs_read(NULL, &read_req, 0, NULL, 0, -1, NULL);
```

Comment:
hum, what should happen now if someone passes a NULL pointer then?

Reply:
I guess nothing? Since uv_fs_cleanup would also do nothing if a NULL pointer is passed?

Reply:
It crashes because none of the req init logic checks for `NULL`. That would happen for other calls too though. I think we should either crash consistently or update the init logic to handle `NULL`.

Reply:
So the more I think about this, the more I think we should just assert on req for these functions. If we return `UV_EINVAL` in these cases, there is no way for the consumer to know if they can cleanup the request or not. It also seems pretty nonsensical to call any of these without a request.

Reply:
@saghul I added a second commit to support `NULL` requests for all of the fs functions. The new logic is "if the request is `NULL`, immediately return `UV_EINVAL`. Otherwise, initialize the request before doing anything else." I think that should allow all requests to be cleaned up. PTAL

Reply:
Please also add some tests which use a NULL req.